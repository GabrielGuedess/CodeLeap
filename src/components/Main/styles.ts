import styled from 'styled-components';

export const Wrapper = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  width: 100%;
  height: 100%;
  padding: 3rem;

  color: #fff;
  text-align: center;

  background-color: #06092b;
`;

export const Logo = styled.img`
  width: 25rem;
  margin-bottom: 2rem;
`;

export const Title = styled.h1`
  margin-bottom: 1rem;
  font-size: 2.5rem;
`;

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`;

export const Illustration = styled.img`
  width: min(30rem, 100%);
  margin-top: 3rem;
`;
