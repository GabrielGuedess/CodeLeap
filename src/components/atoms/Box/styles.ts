import styled, { css } from 'styled-components';

import { type BoxProps } from '.';

const wrapperModifiers = {
  small: () => css`
    width: 50rem;
  `,
  normal: () => css`
    width: 100%;
  `,
  hasPadding: () => css`
    display: flex;
    flex-direction: column;
    gap: 2.4rem;
    padding: 2.4rem;
  `,
};

export const Wrapper = styled.div<Pick<BoxProps, 'size' | 'hasPadding'>>`
  ${({ theme, size, hasPadding }) => css`
    overflow: hidden;
    background: ${theme.colors.white};
    border: 1px solid ${theme.colors.border};
    border-radius: 1.6rem;

    ${wrapperModifiers[size!]}
    ${hasPadding && wrapperModifiers.hasPadding()}
  `}
`;
