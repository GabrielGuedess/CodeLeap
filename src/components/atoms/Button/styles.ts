import { type DefaultTheme } from 'styled-components';
import styled, { css } from 'styled-components';

import { grayscale, darken } from 'polished';

type ButtonProps = {
  isLoading: boolean;
  variant: 'primary' | 'secondary' | 'alert' | 'outline';
};

const buttonModifiers = {
  primary: (theme: DefaultTheme, isLoading: boolean) => css`
    background: ${isLoading
      ? darken(0.1, theme.colors.primary)
      : theme.colors.primary};

    :not(:disabled):is(:hover, :focus) {
      background: ${darken(0.1, theme.colors.primary)};
    }

    :disabled {
      cursor: not-allowed;
      background: ${grayscale(theme.colors.primary)};
    }
  `,
  secondary: (theme: DefaultTheme, isLoading: boolean) => css`
    background: ${isLoading
      ? darken(0.1, theme.colors.green)
      : theme.colors.green};

    :not(:disabled):is(:hover, :focus) {
      background: ${darken(0.1, theme.colors.green)};
    }

    :disabled {
      cursor: not-allowed;
      background: ${grayscale(theme.colors.green)};
    }
  `,
  alert: (theme: DefaultTheme, isLoading: boolean) => css`
    background: ${isLoading ? darken(0.1, theme.colors.red) : theme.colors.red};

    :not(:disabled):is(:hover, :focus) {
      background: ${darken(0.1, theme.colors.red)};
    }

    :disabled {
      cursor: not-allowed;
      background: ${grayscale(theme.colors.red)};
      box-shadow: ${theme.colors.red} 0px 0px 0px 1.5px;
    }
  `,
  outline: (theme: DefaultTheme, isLoading: boolean) => css`
    color: ${theme.colors.primaryText};
    background: transparent;
    border: 1px solid
      ${isLoading ? theme.colors.primaryText : theme.colors.borderButton};
    border-radius: 8px;

    :not(:disabled):is(:hover, :focus) {
      border: 1.5px solid ${theme.colors.primaryText};
    }
  `,
};

export const Wrapper = styled.button<ButtonProps>`
  ${({ theme, isLoading, variant }) => css`
    cursor: pointer;

    display: flex;
    align-items: center;
    justify-content: center;

    width: 12rem;
    height: 3.2rem;

    font-size: 1.6rem;
    font-weight: 700;
    color: ${theme.colors.white};

    border: none;
    border-radius: 0.8rem;

    transition: ${theme.transition.default};

    ${buttonModifiers[variant!](theme, isLoading)}
  `}
`;
