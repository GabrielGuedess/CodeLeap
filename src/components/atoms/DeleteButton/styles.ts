import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    cursor: pointer;
    color: ${theme.colors.white};
  `}
`;
