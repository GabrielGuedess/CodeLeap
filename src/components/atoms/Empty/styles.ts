import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  padding: 2.4rem;
`;

export const Title = styled.h3`
  ${({ theme }) => css`
    margin-bottom: 2.4rem;

    font-size: 2.2rem;
    font-weight: 700;
    color: ${theme.colors.primaryText};
    text-align: center;
  `}
`;
