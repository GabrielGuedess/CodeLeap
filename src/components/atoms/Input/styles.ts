import styled, { css } from 'styled-components';

import * as LabelRadix from '@radix-ui/react-label';

type InputProps = {
  hasError: boolean;
};

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  gap: 0.8rem;
`;

export const Label = styled(LabelRadix.Root)`
  font-size: 1.6rem;
`;

export const Input = styled.input<InputProps>`
  ${({ theme, hasError }) => css`
    width: 100%;
    padding: 0.8rem 1.12rem;

    font-size: 1.4rem;

    border: 1px solid ${hasError ? theme.colors.red : theme.colors.inputBorder};
    border-radius: 0.8rem;

    :focus {
      box-shadow: ${theme.colors.boxShadow} 0px 0px 0px 1.5px;
    }

    ::placeholder {
      color: ${theme.colors.secondaryText};
    }
  `}
`;

export const Error = styled.span`
  ${({ theme }) => css`
    font-size: 1.2rem;
    color: ${theme.colors.red};
  `}
`;
