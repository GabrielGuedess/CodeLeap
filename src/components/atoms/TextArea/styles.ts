import styled, { css } from 'styled-components';

import * as LabelRadix from '@radix-ui/react-label';

type TextAreaProps = {
  hasError: boolean;
};

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  gap: 0.8rem;
`;

export const Label = styled(LabelRadix.Root)`
  font-size: 1.6rem;
`;

export const TextArea = styled.textarea<TextAreaProps>`
  ${({ theme, hasError }) => css`
    resize: vertical;

    width: 100%;
    min-height: 7.4rem;
    max-height: 26rem;
    padding: 0.8rem 1.12rem;

    font-size: 1.4rem;

    border: 1px solid ${hasError ? theme.colors.red : theme.colors.inputBorder};
    border-radius: 0.8rem;

    :focus {
      box-shadow: ${theme.colors.boxShadow} 0px 0px 0px 1.5px;
    }

    ::placeholder {
      color: ${theme.colors.secondaryText};
    }
  `}
`;

export const Error = styled.span`
  ${({ theme }) => css`
    font-size: 1.2rem;
    color: ${theme.colors.red};
  `}
`;
