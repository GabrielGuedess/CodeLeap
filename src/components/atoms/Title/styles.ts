import styled, { css } from 'styled-components';

import { type TitleProps } from '.';

export const Title = styled.h2<Pick<TitleProps, 'color'>>`
  ${({ theme, color }) => css`
    font-size: 2.2rem;
    font-weight: 700;
    color: ${color === 'primary'
      ? theme.colors.primaryText
      : theme.colors.white};
  `}
`;
