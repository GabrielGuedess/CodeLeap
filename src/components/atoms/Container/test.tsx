import { renderWithTheme } from 'utils/tests/helpers';

import { theme } from 'styles/theme';

import { Container } from '.';

describe('<Container />', () => {
  it('should render the container correctly', () => {
    // Arrange
    const { container } = renderWithTheme(
      <Container>
        <span>CodeLeap</span>
      </Container>,
    );

    // Assert
    expect(container.firstChild).toHaveStyleRule(
      'max-width',
      theme.grid.container,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
