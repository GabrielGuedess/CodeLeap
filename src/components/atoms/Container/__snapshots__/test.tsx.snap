// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render the container correctly 1`] = `
.c0 {
  max-width: 80rem;
  margin: 0 1.6rem;
}

@media (min-width:832px) {
  .c0 {
    margin-left: auto;
    margin-right: auto;
  }
}

<div
  class="c0"
>
  <span>
    CodeLeap
  </span>
</div>
`;
